var/spack/repos/builtin/packages/mct/package.py

```diff
@@ -0,0 +1,26 @@
+# Copyright 2013-2023 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack.package import *
+
+
+class Mct(AutotoolsPackage):
+    """Model Coupling Toolkit (MCT): toolkit to support the construction
+    of highly portable and extensible high-performance couplers for
+    distributed memory parallel coupled models."""
+
+    homepage = "https://github.com/MCSclimate/MCT"
+    url = "https://github.com/MCSclimate/MCT/archive/refs/tags/MCT_2.11.0.tar.gz"
+
+    maintainers("climbfuji")
+
+    # This is a custom license that doesn't match any of the ones listed
+    # in https://spdx.org/licenses/, but similar to Creative Commons?
+    license("https://github.com/MCSclimate/MCT/blob/master/LICENSE")
```

Comment:
@boomanaiden154 @tgamblin @alecbcs did y'all discuss what to do about cases where a package has a unique license that does not have an SPDX identifier? Should such packages just not include `license()` for now? I didn't see any conversation about licenses without SPDX identifiers in that PR or in the docs added by it.

For uncommon licenses, this seems reasonable. I don't see any examples of `license(URL)` though: I looked at https://github.com/spack/spack/pull/39346 to see if there were any mentions of one-off licenses and didn't see any. I notice that `zlib` has `license("Zlib")` but it turns out that's actually documented at https://spdx.org/licenses/ . Furthermore, https://github.com/spack/spack/pull/39346#issuecomment-1675294385 talks about doing "SPDX validation" which sounds like it would complain about this.

Reply:
There hasn't been a lot of discussion about it yet, but at least my plan was to adopt a couple non-compliant identifiers to handle cases like this, particularly `CUSTOM` for the case where a license doesn't match any of the SPDX ids. We should probably formalize/document what we want to do though.

Leaving it out for now would probably be fine. Putting a url in the field is definitely incorrect.

I'm hoping to have some infrastructure in place to validate that the expression in `license` is a valid SPDX expression, but haven't had the time to write a SPDX parser yet to get that working.